### DISCUSSION TITLE
How to get the line profile values in KAM or GND map?

### DISCUSSION THREAD

**Discussion by Yangchao-D** [mtex-toolbox/mtex#1917] [Ask Anything]
Hi all,

I was trying to analyze the plastic deformation gradient in my material, so I used KAM or GND map to have a quick look. What I want to check are certain areas. Do you have any suggestions?

The issue with the line profile is that I can only plot profiles for misorientation angles to the specific point. See https://mtex-toolbox.github.io/EBSDProfile.html
So how to get the KAM or GND values in this line from KAM or GND map?

Best wishes,
Yangchao

**Comment by AzdiarGazder** [comment]
Hi @ycjl-deng,

Please click the following [weblink](https://github.com/AzdiarGazder/mtexTools/tree/main/lineProfile) to a script that draws line profiles within MTEX.
Hope this helps.

Warm regards,
Azdi

**Reply by pj-decastro** [reply]
Hello @AzdiarGazder ,

I am interested in the same solution as the creater of this thread (line analysis of KAM values). Therefore, I have incorporated your Line-Profile codes in Matlab. Thank you so much for these neat functionalities!

Nevertheless, I can not make it work with some data. For instance, by investigating the already available 'Twins' or 'Forsterite' datasets it seems to work normally. I could plot the band contrast variations and so on. However, with datasets containing other properties, as the confidence index or image quality, the script seems not to work.

This is the reported error:
"Unable to perform assignment because the left and right sides have a different number of elements.
Error in EBSD/lineProfile (line 102)
    gebsdProperty(~isnan(idxMatrix)) = varargin{1}; % replace numeric values into the NaN array"

I have also tried to execute the same function to plot the orientations. Here, we get another error:

"Index exceeds the number of array elements (1).
Error in EBSD/lineProfile (line 217)
                    angle(ebsdLine(1:end-1).orientations,ebsdLine(2:end).orientations)/degree,..."

When I transform line 216 to 219, it seems the script works.

Thank you again,
Pedro

**Comment by AzdiarGazder** [comment]
Hi @pj-decastro,

You're very welcome to the new functionalities.

And thank you for pointing out the limitations with lineProfile.
I have updated the function [**which can be found here**](https://github.com/AzdiarGazder/mtexTools/tree/main/lineProfile).
I have also included a couple of new examples for users.

Hope this helps! :)

Warm regards,
Azdi

**Comment by kilir** [comment]
Hi,
an easy way using Mtex is to write the variable into the ebsd proeprty `ebsd.prop` and then simply plot it as a function of distance. No need for anything else.

Example:
```
mtexdata titanium
ebsd.prop.kam = ebsd.KAM('threshold',10*degree);
plot(ebsd,ebsd.prop.kam)
xy = ginput(2) % click start an stop of your line
%%
[ebsdline, dist] = spatialProfile(ebsd,xy);
% plot kam along the line
plot(dist,ebsdline.prop.kam/degree)
```
Cheers,
Rüdiger

**Reply by LuqingCui** [reply]
Hi Rüdiger,

Thank you so much for these neat functionalities! The ebsd proeprty ebsd.prop works for the KAM line profile.
I am also interested in the line profile values in the GND maps. 
Could you please provide the code using ebsd property (ebsd.prop) for the line profile values in GND maps?

Best regards,
Luqing Cui

**Reply by kilir** [reply]
Hi,
well, compute the GND (https://mtex-toolbox.github.io/GND.html) and assign it to something like `ebsd.prop.GND`, exactly the same way as above for the kam.

Hope that helps.

Cheers,
Rüdiger

**Reply by LuqingCui** [reply]
Hi Rüdiger,

Thanks for your reply! 
Now I have got the GND map using the code here (https://mtex-toolbox.github.io/GND.html) as follows.
![image](https://github.com/mtex-toolbox/mtex/assets/72967420/7d71fafd-caa7-4e06-a653-e1172a71b382)

Is the code below correct? I think the first and last lines are Incorrect.
**ebsd.prop.gnd = ebsd.GND('threshold',10*degree);**
plot(ebsd,ebsd.prop.gnd)
xy = ginput(2) % click start an stop of your line
%%
[ebsdline, dist] = spatialProfile(ebsd,xy);
% plot gnd along the line
**plot(dist,ebsdline.prop.gnd/degree)**

Best regards,
Luqing Cui

**Reply by kilir** [reply]
Hi,
the `'threshold'` does not have any meaning there, but if it does what you want and expect , it's ok.
Cheers,
Rdiger

**Reply by wangfanglei2** [reply]
你好，请问我用官网上的gnd代码绘制出的图，晶界处会有白色空缺，官网的示例也会有，请问您是如何避免的呢？

> Hi Rüdiger,
> 
> Thanks for your reply! Now I have got the GND map using the code here (https://mtex-toolbox.github.io/GND.html) as follows. ![image](https://private-user-images.githubusercontent.com/72967420/284758273-7d71fafd-caa7-4e06-a653-e1172a71b382.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNDE2NTcsIm5iZiI6MTczNjM0MTM1NywicGF0aCI6Ii83Mjk2NzQyMC8yODQ3NTgyNzMtN2Q3MWZhZmQtY2FhNy00ZTA2LWE2NTMtZTExNzJhNzFiMzgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA4VDEzMDIzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmMmQ0NmIyMzMyZWJhMmU4NzA4ZWVmYWMzOTRlZTBjYTU0NmQ0NDFiYzE5MTRiM2YwMTRiNmE4OTU4MmQ5ZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Jkxio7pFQ1vE9AR9kqFMwiLw3AgwwcF4-Iy7tJxUfxQ)↳
> 
> Is the code below correct? I think the first and last lines are Incorrect. **ebsd.prop.gnd = ebsd.GND('threshold',10*degree);** plot(ebsd,ebsd.prop.gnd) xy = ginput(2) % click start an stop of your line %% [ebsdline, dist] = spatialProfile(ebsd,xy); % plot gnd along the line **plot(dist,ebsdline.prop.gnd/degree)**↳
> 
> Best regards, Luqing Cui↳
